scripts/test/SANS/state/state_test.py

```diff
@@ -128,33 +125,9 @@ def check_bad_and_good_values(self, bad_state, good_state):
         state = self._get_state(good_state)
         self.assertIsNone(state.validate())
 
-    def test_that_raises_when_move_has_not_been_set(self):
-        self.check_bad_and_good_values({"move": None}, {"move": MockStateMove()})
-
-    def test_that_raises_when_reduction_has_not_been_set(self):
-        self.check_bad_and_good_values({"reduction": None}, {"reduction": MockStateReduction()})
-
-    def test_that_raises_when_slice_has_not_been_set(self):
-        self.check_bad_and_good_values({"slice": None}, {"slice": MockStateSliceEvent()})
-
-    def test_that_raises_when_mask_has_not_been_set(self):
-        self.check_bad_and_good_values({"mask": None}, {"mask": MockStateMask()})
-
-    def test_that_raises_when_wavelength_has_not_been_set(self):
-        self.check_bad_and_good_values({"wavelength": None}, {"wavelength": MockStateWavelength()})
-
-    def test_that_raises_when_save_has_not_been_set(self):
-        self.check_bad_and_good_values({"save": None}, {"save": MockStateSave()})
-
-    def test_that_raises_when_scale_has_not_been_set(self):
-        self.check_bad_and_good_values({"scale": None}, {"scale": MockStateScale()})
-
-    def test_that_raises_when_adjustment_has_not_been_set(self):
-        self.check_bad_and_good_values({"adjustment": None}, {"adjustment": MockStateAdjustment()})
-
-    def test_that_raises_when_convert_to_q_has_not_been_set(self):
-        self.check_bad_and_good_values({"convert_to_q": None}, {"convert_to_q": MockStateConvertToQ()})
-
+    # TODO reimplement validator tests in new state objects
```

Comment:
At some point yes, as we have to validate as we execute which isn't brilliant

The validators were removed out of most State* objects since we need to create them with an inconsistent state as the user fills in various parameters. Previously it expected the objects to be packed in a certain order which was great for the legacy parser, but won't work if you try to do it outside builders in a single atomic step (i.e. incrementally)

IMO if we have a request from the SANS team / various users are running into obscure messages we can look at adding them back. I'm thinking the state objects stay dumb and then we have a `StateXValidator` which then checks it, so that we have a clearly defined point everything gets validated.
However, this is going to take significant work, and the underlying algorithm validation tends to do a good job at giving sane error messages

Reply:
Is it worth creating an issue for that? I'm a bit worried if this TODO is still there in a year or two's time it won't mean a lot to future developers!

Reply:
I've removed the TODO, I'm tempted to err against opening an issue? Since I suspect it will likely end up going stale or it will end up user reported sooner rather than later

Reply:
It depends if you think it's worth capturing some context for future devs, even if we close it as stale... your call!

Reply:
I've opened #29241 for this, it should document my thoughts and why were here a bit better.
This should also cover us for the inevitable, "well it worked before....", question too